src/main/java/com/exasol/glue/FilterConverter.java

```diff
@@ -0,0 +1,203 @@
+
+import org.apache.spark.sql.sources.*;
+
+/**
+ * A class that converts Spark {@link Filter} conditions into Exasol {@link ValueExpression}.
+ */
+public final class FilterConverter {
+    private static final Map<Class<? extends Filter>, OperationType> FILTERS = getMappings();
+
+    /**
+     * Converts an array of Spark {@link Filter} conditions into Exasol SQL expression.
+     *
+     * @param filters array of filters
+     * @return optional {@link ValueExpression}
+     */
+    public Optional<BooleanExpression> convert(final Filter[] filters) {
+        if (filters == null || filters.length == 0) {
+            return Optional.empty();
+        }
+        final List<BooleanExpression> expressions = new ArrayList<>();
+        for (final Filter filter : filters) {
+            final BooleanExpression convertedResult = convertFilter(filter);
+            if (convertedResult == null) {
+                return Optional.empty();
+            }
+            expressions.add(convertedResult);
+        }
+        return Optional.of(BooleanTerm.and(expressions.toArray(new BooleanExpression[] {})));
+    }
+
+    /**
+     * Checks is {@link Filter} filter can be converted into Exasol expression.
+     *
+     * @param filter filter to check
+     * @return {@code true} if filter can be converted; {@code false} otherwise
+     */
+    public boolean isFilterSupported(final Filter filter) {
+        return convertFilter(filter) != null;
+    }
+
+    private BooleanExpression convertFilter(final Filter filter) {
+        final OperationType operationType = FILTERS.getOrDefault(filter.getClass(), OperationType.UNKNOWN);
+        switch (operationType) {
+        case EQUAL_TO:
+            final EqualTo equalTo = (EqualTo) filter;
+            return BooleanTerm.eq(column(equalTo.attribute()), getLiteralValue(equalTo.value()));
+
+        case GREATER_THAN:
+            final GreaterThan greaterThan = (GreaterThan) filter;
+            return BooleanTerm.gt(column(greaterThan.attribute()), getLiteralValue(greaterThan.value()));
+
+        case GREATER_THAN_OR_EQUAL:
+            final GreaterThanOrEqual greaterThanOrEqual = (GreaterThanOrEqual) filter;
+            return BooleanTerm.ge(column(greaterThanOrEqual.attribute()), getLiteralValue(greaterThanOrEqual.value()));
+
+        case LESS_THAN:
+            final LessThan lessThan = (LessThan) filter;
+            return BooleanTerm.lt(column(lessThan.attribute()), getLiteralValue(lessThan.value()));
+
+        case LESS_THAN_OR_EQUAL:
+            final LessThanOrEqual lessThanOrEqual = (LessThanOrEqual) filter;
+            return BooleanTerm.le(column(lessThanOrEqual.attribute()), getLiteralValue(lessThanOrEqual.value()));
+
+        case STRING_STARTS_WITH:
+            final StringStartsWith startsWith = (StringStartsWith) filter;
+            return BooleanTerm.like(column(startsWith.attribute()), stringLiteral(startsWith.value() + "%"));
```

Comment:
I think we should escape the `startsWith.value()` here. Otherwise, a filter that contains a `%` could produce wrong results. Or even be a security risk.
If you escape, make sure you pass an escape char to the LIKE function, since users can change the default escape char.
https://docs.exasol.com/db/latest/sql_references/predicates/not_like.htm

Reply:
same for other string functions